internal/operands/tekton-cleanup/reconcile.go

```diff
@@ -0,0 +1,195 @@
+package tekton_cleanup
+
+import (
+	"fmt"
+
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ssp "kubevirt.io/ssp-operator/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+)
+
+// +kubebuilder:rbac:groups=tekton.dev,resources=pipelines,verbs=list;watch;create;update
+// +kubebuilder:rbac:groups=tekton.dev,resources=tasks,verbs=list;watch;update
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=list;watch;update
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;rolebindings,verbs=list;watch;update
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=list;watch;create;update
+
+// Below are RBAC for deployed ClusterRoles. We still need these permissions so we can update annotations on existing ClusterRoles
+
+// +kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;update;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=create
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;create;patch
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances;virtualmachines,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachines/finalizers,verbs=get
+// +kubebuilder:rbac:groups=cdi.kubevirt.io,resources=datasources,verbs=get;create;delete
+// +kubebuilder:rbac:groups=subresources.kubevirt.io,resources=virtualmachines/restart;virtualmachines/start;virtualmachines/stop,verbs=update
+// +kubebuilder:rbac:groups=template.openshift.io,resources=processedtemplates,verbs=create
+// +kubebuilder:rbac:groups=template.openshift.io,resources=templates,verbs=get;list;watch;create;update;delete
```

Comment:
Most of these can be removed now right?

Reply:
No, these rules are still needed, because the current code updates an existing `ClusterRole` objects.
The operator needs to have all the permissions that are in the roles to be allowed to update them.